06.wc/my_wc.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = ARGV.getopts('l')
+  contents = read_contents
+  display_result(contents, options)
+end
+
+def display_result(contents, options)
+  total_lines = 0
+  total_words = 0
+  total_bytes = 0
+  contents.each do |file_name, content|
+    line_count = content.count
+    total_lines += line_count
+    word_count = content.join.split(/\s+/).count
+    total_words += word_count
+    byte_count = content.join.length
+    total_bytes += byte_count
+    puts adjust_lines_words_bytes(line_count, word_count, byte_count, file_name, options)
+  end
+
+  return if contents.size == 1
+
+  puts adjust_lines_words_bytes(total_lines, total_words, total_bytes, 'total', options)
+end
+
+def adjust_lines_words_bytes(lines, words, bytes, file_or_total, options)
+  result = +''
+  result << format_number(lines)
+  result << "#{format_number(words)}#{format_number(bytes)}" unless options['l']
```

Comment:
行数は多少増えますが、個人的にはこう書いた方が一貫性があって読みやすい気がします〜。



```suggestion
  unless options['l']
    result << "#{format_number(words)}"
    result << "#{format_number(bytes)}"
  end
```